smart-contracts/wasm-transform/src/artifact.rs

```diff
@@ -21,16 +21,19 @@ use std::{
 
 #[derive(Copy, Clone)]
 /// Either a short or long integer.
+/// The reason this is a union is that after Wasm module validation we are
+/// guaranteed that the program is well typed. Since all instructions have
+/// clear, fixed types, we can determine from the instruction which value we
+/// expect on the stack. Using a union saves on the discriminant compared to
+/// using an enum, leading to 50% less space used on the stack, as well as
+/// removes the need to handle impossible cases.
 pub union StackValue {
     pub short: i32,
     pub long:  i64,
 }
 
 impl std::fmt::Debug for StackValue {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.write_str("StackValue")
-        // write!(f, "{}", unsafe { self.short })
-    }
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result { f.write_str("StackValue") }
```

Comment:
This just prints "StackValue". Is that sufficient for what we need?

Reply:
Well it would be better if it printed the actual value. But that's not possible since it's a union so we don't know which value it is. I could always display assuming it is a big value, but that would be wrong in general.